src/Team-Services-Bot.Api/Controllers/MessagesController.cs

```diff
@@ -25,16 +26,23 @@ public class MessagesController : ApiController
         /// </summary>
         public async Task<HttpResponseMessage> Post([FromBody]Activity activity)
         {
-            if (activity.Type == ActivityTypes.Message)
+            try
             {
-                await Conversation.SendAsync(activity, () => new Dialogs.RootDialog());
+                if (activity.Type == ActivityTypes.Message)
+                {
+                    await Conversation.SendAsync(activity, () => new Dialogs.RootDialog());
+                }
+                else
+                {
+                    HandleSystemMessage(activity);
+                }
+                var response = Request.CreateResponse(HttpStatusCode.OK);
+                return response;
             }
-            else
+            catch (Exception ex)
             {
-                HandleSystemMessage(activity);
+                throw ex;
```

Comment:
I wonder if it's in the default template so it gets picked up in Visual Studio during development, I noticed that a lot of times bot framework exceptions get 'eaten up' locally and you only catch them if you change the default exception settings in VS. This would also be one the places to make an appInsights.trackException call too to trace that exception. But either way, you can add the catch later once you get to adding app insights, or just leave it as it is now, up to you.